src/ray/rpc/client_call.h

```diff
@@ -195,6 +195,11 @@ class ClientCallManager {
     for (auto &polling_thread : polling_threads_) {
       polling_thread.join();
     }
+    // gRPC's completion queue won't clear our call tags, we need to clear call tags that
+    // are waiting for server's response.
+    for (auto &ptr : allocated_tags) {
```

Comment:
Why don't we just make the client tag unique_ptr?

Reply:
yup prefer RAII (unique_ptr). it will have similar performance as new/delete.

Reply:
Actually, I don't know how to use unique_ptr, could you say more about it?

gRPC asks us to pass a `void*` tag to it. Which won't be managed by unique_ptr.

```cpp
    auto tag = new ClientCallTag(call);
    allocated_tags_.emplace(tag);
    call->response_reader_->Finish(&call->reply_, &call->status_, (void *)tag);
```

Reply:
Yeah, I agree with you. Committed.